kubernetes-client/src/main/java/io/fabric8/kubernetes/client/BaseClient.java

```diff
@@ -94,20 +94,24 @@ public Config getConfiguration() {
   }
 
   @Override
-  public <C extends Client> Boolean isAdaptable(Class<C> type) {
+  public <C> Boolean isAdaptable(Class<C> type) {
     ExtensionAdapter<C> adapter = Adapters.get(type);
     if (adapter != null) {
       return adapter.isAdaptable(this);
-    } else {
+    } else if (type.isAssignableFrom(OkHttpClient.class)) {
```

Comment:
Add inner class (OkHttpClientExtensionAdapter) to BaseClient (or where httpClient field is),
and register it against Adapters in BaseClient ctor -- no reflection here.